File: content/problems/0682-baseball-game/solutions/score-stack/Solution.java
class Solution {
    public int calPoints(String[] operations){
        List<Integer> scores=new ArrayList<>();
        for(String operation:operations){
            if(operation.equals("C"))scores.remove(scores.size()-1);
            else if(operation.equals("D"))scores.add(scores.get(scores.size()-1)*2);
            else if(operation.equals("+"))scores.add(scores.get(scores.size()-1)+scores.get(scores.size()-2));
            else scores.add(Integer.parseInt(operation));
        }
        int total=0;
        for(int score:scores)total+=score;
        return total;
    }
}

File: content/problems/0682-baseball-game/solutions/score-stack/Solution.kt
class Solution {
    fun calPoints(operations:Array<String>):Int{
        val scores=mutableListOf<Int>();
        for(operation in operations)when(operation){
            "C"->scores.removeAt(scores.lastIndex);
            "D"->scores.add(scores.last()*2);
            "+"->scores.add(scores.last()+scores[scores.lastIndex-1]);
            else->scores.add(operation.toInt())
        };
        return scores.sum()
    }
}
